frontend/src/pages/ExperimentDetails.tsx

```diff
@@ -326,19 +336,71 @@ export class ExperimentDetails extends Page<{}, ExperimentDetailsState> {
         logger.error(`Error fetching recurring runs for experiment: ${experimentId}`, err);
       }
 
-      this.setStateSafe({ activeRecurringRunsCount, experiment });
+      this.setStateSafe({
+        activeRecurringRunsCount,
+        experiment,
+        runStorageState,
+      });
+      this._selectionChanged([]);
     } catch (err) {
       await this.showPageError(`Error: failed to retrieve experiment: ${experimentId}.`, err);
       logger.error(`Error loading experiment: ${experimentId}`, err);
     }
   }
 
+  /**
+   * Users can choose to show runs list in different run storage states. After finish changing storage state, call cb to notify child component for new props.
+   *
+   * @param tab selected by user for run storage state
+   * @param cb callback to notify child component
+   */
+  private _onRunTabSwitch(tab: RunListsGroupTab): void {
+    let runStorageState = RunStorageState.AVAILABLE;
+    if (tab === RunListsGroupTab.ARCHIVE) {
+      runStorageState = RunStorageState.ARCHIVED;
+    }
+    this.setStateSafe(
+      {
+        runStorageState,
+      },
+      () => {
+        this._selectionChanged([]);
+        this.refreshRunListsRouter();
```

Comment:
I might be missing some context --
When storageState changes, I think RunList should automatically refresh. Why do we need to trigger manually?

Reply:
It is because the selectionChanged can always be empty when switching tabs, then there is nothing to refresh in child component `RunList` because props for it doesn't change, React will only update the component that are affected, but nothing else. I have updated to use one-way data flow.